template.yaml

```diff
@@ -567,67 +563,48 @@ Resources:
-  #        HostedZoneId: !GetAtt FrontCustomDomain.RegionalHostedZoneId
-  #        EvaluateTargetHealth: false
-
-  #  FrontApiMapping:
-  #    Type: AWS::ApiGatewayV2::ApiMapping
-  #    Properties:
-  #      DomainName: !FindInMap [EnvironmentVariables, !Ref Environment, FEURL ]
-  #      ApiId: !Ref ApiGwHttpEndpoint
-  #      Stage: "$default"
-  #    DependsOn:
-  #      - FrontCustomDomain
-
-
-  #  # app domain
-  #  AppFrontCustomDomain:
-  #    Type: AWS::ApiGatewayV2::DomainName
-  #    Properties:
-  #      DomainName: !FindInMap [EnvironmentVariables, !Ref Environment, APPURL ]
-  #      DomainNameConfigurations:
-  #        - CertificateArn: !Sub "{{resolve:ssm:/${Environment}/Platform/ACM/PrimaryZoneWildcardCertificateARN}}"
-  #          EndpointType: REGIONAL
-  #          SecurityPolicy: TLS_1_2
-
-  #  AppFrontApiRecord:
-  #    Type: AWS::Route53::RecordSet
-  #    Properties:
-  #      Name: !FindInMap [EnvironmentVariables, !Ref Environment, APPURL ]
-  #      Type: A
-  #      HostedZoneId: !Sub "{{resolve:ssm:/${Environment}/Platform/Route53/PrimaryZoneID}}"
-  #      AliasTarget:
-  #        DNSName: !GetAtt AppFrontCustomDomain.RegionalDomainName
-  #        HostedZoneId: !GetAtt AppFrontCustomDomain.RegionalHostedZoneId
-  #        EvaluateTargetHealth: false
-
-  #  AppFrontApiMapping:
-  #    Type: AWS::ApiGatewayV2::ApiMapping
-  #    Properties:
-  #      DomainName: !FindInMap [EnvironmentVariables, !Ref Environment, APPURL ]
-  #      ApiId: !Ref ApiGwHttpEndpoint
-  #      Stage: "$default"
-  #    DependsOn:
-  #      - AppFrontCustomDomain
+  F2FFrontCustomDomain:
+    Type: AWS::ApiGatewayV2::DomainName
+    Properties:
+      DomainName: !If
+        - IsNotDevelopment
+        - !Sub
+          - "www.${DNSSUFFIX}"
+          - DNSSUFFIX: !FindInMap [EnvironmentVariables, !Ref Environment, DNSSUFFIX]
+        - !Sub
+          - "${AWS::StackName}.${DNSSUFFIX}"
+          - DNSSUFFIX: !FindInMap [EnvironmentVariables, !Ref Environment, DNSSUFFIX]
+      DomainNameConfigurations:
+        - CertificateArn: !Sub "{{resolve:ssm:/${Environment}/Platform/ACM/PrimaryZoneWildcardCertificateARN}}"
+          EndpointType: REGIONAL
+          SecurityPolicy: TLS_1_2
+
+  F2FFrontApiRecord:
+    Type: AWS::Route53::RecordSet
+    Properties:
+      Name: !If
+        - IsNotDevelopment
+        - !Sub
+          - "www.${DNSSUFFIX}"
+          - DNSSUFFIX: !FindInMap [EnvironmentVariables, !Ref Environment, DNSSUFFIX]
+        - !Sub
+          - "${AWS::StackName}.${DNSSUFFIX}"
+          - DNSSUFFIX: !FindInMap [EnvironmentVariables, !Ref Environment, DNSSUFFIX]
+      Type: A
+      HostedZoneId: !ImportValue PublicHostedZoneId
+      AliasTarget:
+        DNSName: !GetAtt F2FFrontCustomDomain.RegionalDomainName
+        HostedZoneId: !GetAtt F2FFrontCustomDomain.RegionalHostedZoneId
+        EvaluateTargetHealth: false
+
+  F2FFrontApiMapping:
+    Type: AWS::ApiGatewayV2::ApiMapping
```

Comment:
does it work well without that `DependsOn` (worth testing by creating a new one/modifying an existing one)?

Reply:
Yes - since the ApiId and Stage have a reference to the APIGW and Stage this creates the implicit dependency